### PR TITLE
implemented TDD and integration tests for CartItem and ShoppingCart e…

### PR DESCRIPTION
**Pull request by s-hrynus**
…ntities
coverage

### CODE REVIEW

File: src/test/java/mate/academy/controller/ShoppingCartControllerIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+        cartItemRequestDto = new CartItemRequestDto(book.getId(),DEFAULT_QUANTITY);
+
+        invalidRequestDto = new CartItemRequestDto(INVALID_ID, DEFAULT_QUANTITY);
+
+        cartItemResponseDto = new CartItemResponseDto();
+        cartItemResponseDto.setId(cartItem.getId());
+        cartItemResponseDto.setBookId(cartItem.getBook().getId());
+        cartItemResponseDto.setQuantity(cartItem.getQuantity());
+        cartItemResponseDto.setBookTitle(book.getTitle());
+    }
```
Comment (olekskov) [inline src/test/java/mate/academy/controller/ShoppingCartControllerIntegrationTest.java]: A lot of boilerplate code in the beginning of test class looks not good.
Also same situation and similar models are in next test class.
Please extract separate methods for standard models creation.
There are lot of options:
1) make parent class (e.g. IntegrationTest) with methods getDefaultCartItem(), getDefaultBook() etc and use in child classes
2) make TestDataUtil or something like that and make there public static methods getDefaultCartItem(), getDefaultBook() etc and reuse in both classes
3) make them static in one test class in reuse in other one

probably some other options if think more about it

**Review by olekskov — CHANGES_REQUESTED** [review]

File: src/test/java/mate/academy/controller/ShoppingCartControllerIntegrationTest.java (hunk 1 above)
Comment (s-hrynus) [reply src/test/java/mate/academy/controller/ShoppingCartControllerIntegrationTest.java]: thx for feedback :)

File: src/test/java/mate/academy/controller/ShoppingCartControllerIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+        user = TestDataUtil.getDefaultUser();
+        book = TestDataUtil.getDefaultBook();
+        cartItem = TestDataUtil.getDefaultCartItem();
+        cartItemRequestDto = TestDataUtil.getDefaultCartItemRequestDto();
+        cartItemResponseDto = TestDataUtil.getDefaultCartItemResponseDto();
+        invalidRequestDto = TestDataUtil.getBookInvalidRequestDto();
+        updateRequestDto = TestDataUtil.getDefaultUpdateRequestDto();
+        shoppingCart = TestDataUtil.getDefaultShoppingCart();
+        shoppingCartDto = TestDataUtil.getDefaultShoppingCartDto();
```
Comment (olekskov) [inline src/test/java/mate/academy/controller/ShoppingCartControllerIntegrationTest.java]: I'd also made static imports for these methods to omit name of class

**Review by olekskov — APPROVED** [review]

File: src/test/java/mate/academy/controller/ShoppingCartControllerIntegrationTest.java (hunk 2 above)
Comment (s-hrynus) [reply src/test/java/mate/academy/controller/ShoppingCartControllerIntegrationTest.java]: Many thanks, I'll do it!

**Pull request merged**